component_setup_cardinal.ipynb

```diff
@@ -0,0 +1,81 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "25lVf3PwsmmV"
+      },
+      "source": [
+        "## Setup Cardinal"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6fzwlrSLOF3w"
+      },
+      "source": [
+        "### Install Cardinal"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1H6SWBCTDq5G"
+      },
+      "source": [
+        "This notebook uses [Cardinal](https://cardinal.readthedocs.io/en/latest/), a Rust package to calculate red flags and the coverage of OCDS data."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "6syz0fkkEdgj"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Install { display-mode: \"form\" }\n",
+        "\n",
+        "! curl -sSOL https://github.com/open-contracting/cardinal-rs/releases/download/0.0.5/ocdscardinal-0.0.5-linux-64-bit.zip\n",
+        "! unzip -oj ocdscardinal-0.0.5-linux-64-bit.zip ocdscardinal-0.0.5-linux-64-bit/ocdscardinal\n",
+        "\n",
+        "def cardinal_calculate_coverage(file_name):\n",
+        "    coverage = !./ocdscardinal coverage $file_name\n",
+        "    fields = (\n",
+        "        pd.DataFrame.from_dict(json.loads(coverage[0]), orient=\"index\", columns=[\"count\"])\n",
+        "        .reset_index()\n",
+        "        .rename(columns={\"index\": \"path\"})\n",
+        "    )\n",
+        "    # Leaves only object members\n",
+        "    fields_table = fields[fields.path.str.contains(\"[a-z]$\")].copy()\n",
+        "    fields_table[\"path\"] = fields_table[\"path\"].str.replace(r\"[][]|^/\", \"\", regex=True)\n",
+        "    return fields_table"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "outputs": [],
+      "source": [
+        "# @title Extract the list of available fields { display-mode: \"form\" }\n",
+        "\n",
+        "fields_table = cardinal_calculate_coverage(file_name)"
+      ],
+      "metadata": {
+        "collapsed": false
+      }
+    }
+  ],
+  "metadata": {
+    "colab": {
+      "provenance": []
+    },
+    "kernelspec": {
+      "display_name": "Python 3",
+      "name": "python3"
+    }
+  },
+  "nbformat": 4,
+  "nbformat_minor": 0
```

Comment:
The order of top-level members is usually nbformat, nbformat_minor, metadata, then cells.